src/main/java/de/tum/in/www1/artemis/repository/metrics/ExerciseMetricsRepository.java

```diff
@@ -0,0 +1,99 @@
+package de.tum.in.www1.artemis.repository.metrics;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.util.Set;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO;
+
+/**
+ * Spring Data JPA repository to fetch exercise related metrics.
+ */
+@Profile(PROFILE_CORE)
+@Repository
+public interface ExerciseMetricsRepository extends JpaRepository<Exercise, Long> {
+
+    /**
+     * Get the exercise information for all exercises in a course.
+     *
+     * @param courseId the id of the course
+     * @return the exercise information for all exercises in the course
+     */
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseInformationDTO(e.id, e.shortName, e.title, COALESCE(e.startDate, e.releaseDate), e.dueDate, e.maxPoints, e.class)
+            FROM Exercise e
+            WHERE e.course.id = :courseId
+            """)
+    Set<ExerciseInformationDTO> findAllExerciseInformationByCourseId(@Param("courseId") long courseId);
+
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO(p.exercise.id, AVG(COALESCE(p.lastScore, 0)))
+            FROM ParticipantScore p
+            WHERE p.exercise.id IN :exerciseIds
+            GROUP BY p.exercise.id
+            """)
+    Set<ScoreDTO> findAverageScore(@Param("exerciseIds") Set<Long> exerciseIds);
```

Comment:
I still feel that average score does not capture the whole picture. Maybe we also need to include how many people participated in an exercise out of a total.